### ISSUE TITLE
about iseg2017

### ISSUE THREAD

**Issue by shu-hai** [ellisdg/3DUnetCNN#21]
Hi, ellisdg,
I was wondering what is the dice coefficient you have got for the infant segmentation dataset?
I am a deep learning beginner.
I am now using your default settings. In the first 15 epochs, the dice coefficient seems stable around 0.57.
Whether the learning rate is too small as 0.00001?
Thanks.

**Comment by ellisdg** [comment]
Hi @shu-hai,

I trained it for a week or so and I was able to get up to 0.8, I believe. If the validation score doesn't seem to be improving, you could try increasing the learning rate. You could also try [turning off the data augmentation](https://github.com/ellisdg/3DUnetCNN/blob/master/iseg/train.py#L51). This was a setting I experimented with to try and prevent the network from over-fitting to the small dataset. 15 epochs isn't very much since the dataset is so small, so you could also just wait a bit longer and see if the validation score improves.

Since the model and training/validation splits are all saved to file, you can start and restart the training as much as you like while playing around with the settings.

I got busy, so I haven't looked at it for a couple months. I was hoping to revisit it later as 0.8 isn't very good. From the papers that I have read, the people who are most successful with the brain image segmentation seem to do the training/segmentation in patches. That is they will split up the image into smaller images so that the CNN can then use a smaller amount of memory and train quicker on a graphics card. This allows them to tweak the settings and get feedback a lot quicker than training on a CPU. So splitting up the dataset into patches might be something to try.

Let me know what changes and settings work best for you. If you are able to get stellar scores using any part of this code, be sure to include a link to the code in the submission.

**Comment by shu-hai** [comment]
Hi, @ellisdg,
I am a little confused about the "dice_coef" function. The label for the infant segmentation has three labels {10,150,250}. Is the function OK for multiple labels? I think the dice_coef is for binary labels.

**Comment by ellisdg** [comment]
The labels (10, 150, 250) are binary labels. Each of the labels represents a different type of tissue. Any type of label is by definition a binary label. Either it has that label or it does not.

A 5X1 image with labels:
[ 250 250 10 150 150 ]

Could also be represented as:
[ 0 0 1 0 0 ]
[ 0 0 0 1 1 ]
[ 1 1 0 0 0 ]

By feeding the above matrix into the training, we are teaching the CNN to detect each label.
